src/converters/transactionsConverter.ts

```diff
@@ -61,4 +70,57 @@ export class TransactionsConverter {
     private bufferFromHex(value?: string) {
         return Buffer.from(value || "", "hex");
     }
+
+    public transactionOnNetworkToOutcome(transactionOnNetwork: ITransactionOnNetwork): TransactionOutcome {
+        // In the future, this will not be needed because the transaction, as returned from the API,
+        // will hold the data corresponding to the direct smart contract call outcome (in case of smart contract calls).
+        const legacyResultsParser = new ResultsParser();
+        const callOutcomeBundle = legacyResultsParser.parseUntypedOutcome(transactionOnNetwork);
+        const callOutcome = new SmartContractCallOutcome({
+            function: transactionOnNetwork.function,
+            returnCode: callOutcomeBundle.returnCode.toString(),
+            returnMessage: callOutcomeBundle.returnMessage,
+            returnDataParts: callOutcomeBundle.values,
+        });
+
+        const contractResults = transactionOnNetwork.contractResults.items.map((result) =>
+            this.smartContractResultOnNetworkToSmartContractResult(result),
+        );
+
+        const logs = new TransactionLogs({
+            address: transactionOnNetwork.logs.address.bech32(),
+            events: transactionOnNetwork.logs.events.map((event) => this.eventOnNetworkToEvent(event)),
+        });
+
+        return new TransactionOutcome({
+            logs: logs,
+            smartContractResults: contractResults,
+            directSmartContractCallOutcome: callOutcome,
+        });
+    }
+
+    private smartContractResultOnNetworkToSmartContractResult(
+        resultOnNetwork: IContractResultItem,
+    ): SmartContractResult {
+        return new SmartContractResult({
+            sender: resultOnNetwork.sender.bech32(),
+            receiver: resultOnNetwork.receiver.bech32(),
+            data: Buffer.from(resultOnNetwork.data),
+            logs: new TransactionLogs({
+                address: resultOnNetwork.logs.address.bech32(),
+                events: resultOnNetwork.logs.events.map((event) => this.eventOnNetworkToEvent(event)),
+            }),
+        });
+    }
+
+    private eventOnNetworkToEvent(eventOnNetwork: ITransactionEvent): TransactionEvent {
+        return new TransactionEvent({
+            address: eventOnNetwork.address.bech32(),
+            identifier: eventOnNetwork.identifier,
+            // TODO: Workaround will be fixed soon, once "topics" become Uint8Array[].
```

Comment:
the PR containing this has been merged, so topic is now `UInt8Array[]`

Reply:
Indeed, thanks!